src/Common/src/Interop/Unix/System.Native/Interop.IPAddress.cs

```diff
@@ -33,6 +33,46 @@ public bool IsIPv6
             internal fixed byte Address[MAX_IP_ADDRESS_BYTES]; // Buffer to fit an IPv4 or IPv6 address
             private  uint _isIPv6;                             // Non-zero if this is an IPv6 address; zero for IPv4.
             internal uint ScopeId;                             // Scope ID (IPv6 only)
+
+            public bool Equals(IPAddress other)
+            {
+                int addressByteCount;
+                if (IsIPv6)
+                {
+                    if (!other.IsIPv6)
+                    {
+                        return false;
+                    }
+                    if (ScopeId != other.ScopeId)
+                    {
+                        return false;
+                    }
+
+                    addressByteCount = IPv6AddressBytes;
+                }
+                else
+                {
+                    if (other.IsIPv6)
+                    {
+                        return false;
+                    }
+
+                    addressByteCount = IPv4AddressBytes;
+                }
+
+                fixed (byte* thisAddress = Address)
+                {
+                    for (int i = 0; i < addressByteCount; i++)
+                    {
+                        if (thisAddress[i] != other.Address[i])
```

Comment:
Interesting.  Why is the fixed necessary for this.Address but not for other.Address?  Is it because IPAddress is a struct, and this could be contained inside of an object that's moveable on the heap, but we know that other lives on the stack since it was passed in as a non-ref arg?